### PR TITLE
add Demisto Bot to exceptionally_allowed_substrings

### PR DESCRIPTION
**Pull request by YuvHayun**
## Related Issues
fixes:
[CIAC-11619](https://jira-dc.paloaltonetworks.com/browse/CIAC-11619)
## Description
Fixing an error that was found [here](https://github.com/demisto/content/pull/35989)

### CODE REVIEW

File: demisto_sdk/commands/validate/validators/DS_validators/DS107_is_description_contains_demisto_word.py [hunk 1]
```diff
@@ -43,7 +43,10 @@ def obtain_invalid_content_items(
                     phrases_to_search=["demisto"],
                     ignore_case=True,
                     text=content_item.description_file.file_content,
-                    exceptionally_allowed_substrings=["/demisto/"],  # in URL
+                    exceptionally_allowed_substrings=[
+                        "/demisto/",
+                        "Demisto Bot",
+                    ],  # in URL
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validators/DS_validators/DS107_is_description_contains_demisto_word.py:49]:
```suggestion
                        "/demisto/",  # in URL
                        "Demisto Bot",
                    ],  
```

File: demisto_sdk/commands/validate/validators/DS_validators/DS107_is_description_contains_demisto_word.py [hunk 2]
```diff
@@ -43,7 +43,10 @@ def obtain_invalid_content_items(
                     phrases_to_search=["demisto"],
                     ignore_case=True,
                     text=content_item.description_file.file_content,
-                    exceptionally_allowed_substrings=["/demisto/"],  # in URL
+                    exceptionally_allowed_substrings=[
+                        "/demisto/",
+                        "Demisto Bot",
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validators/DS_validators/DS107_is_description_contains_demisto_word.py:48]: Explain in a word or two why this is here

File: .changelog/4517.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Fixed an issue where DS107 validation (validate that description doesn't contain demisto word) failed on Demisto Bot word.
```
Comment (dorschw) [inline .changelog/4517.yml:2]:
```suggestion
- description: Fixed an issue where DS107 validation (descriptions shouldn't contain the phrase `demisto`) failed when the `Demisto Bot` was mentioned.
```

**Review by dorschw — APPROVED** [review]

File: demisto_sdk/commands/validate/validators/DS_validators/DS107_is_description_contains_demisto_word.py [hunk 4]
```diff
@@ -43,7 +43,10 @@ def obtain_invalid_content_items(
                     phrases_to_search=["demisto"],
                     ignore_case=True,
                     text=content_item.description_file.file_content,
-                    exceptionally_allowed_substrings=["/demisto/"],  # in URL
+                    exceptionally_allowed_substrings=[
+                        "/demisto/",  # in URL
+                        "Demisto Bot",
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/DS_validators/DS107_is_description_contains_demisto_word.py:48]:
```suggestion
                        "Demisto Bot",  # expected pair.
```

**Pull request merged**